internal/pkg/aws/sessions/sessions_test.go

```diff
@@ -183,12 +193,48 @@ func TestProvider_FromProfile(t *testing.T) {
 		require.NoError(t, err)
 
 		// WHEN
-		sess, err := ImmutableProvider().FromProfile("walk-like-an-egyptian")
+		provider := &Provider{
+			sessionValidator: m,
+		}
+
+		sess, err := provider.FromProfile("walk-like-an-egyptian")
 
 		// THEN
 		require.NoError(t, err)
 		require.Equal(t, "us-west-2", *sess.Config.Region)
 	})
+
+	t.Run("session credentials are incorrect", func(t *testing.T) {
+
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		m := mocks.NewMocksessionValidator(ctrl)
+		m.EXPECT().ValidateCredentials(gomock.Any()).Return(credentials.Value{}, context.DeadlineExceeded)
+
+		ogRegion := os.Getenv("AWS_REGION")
+		defer func() {
+			err := restoreEnvVar("AWS_REGION", ogRegion)
+			require.NoError(t, err)
+		}()
+
+		// Since "walk-like-an-egyptian" is (very likely) a non-existent profile, whether the region information
+		// is missing depends on whether the `AWS_REGION` environment variable is set.
+		err := os.Setenv("AWS_REGION", "us-west-2")
+		require.NoError(t, err)
+
+		// WHEN
+		provider := &Provider{
+			sessionValidator: m,
+		}
+
+		sess, err := provider.FromProfile("walk-like-an-egyptian")
+
+		// THEN
+		require.NotNil(t, err)
+		require.EqualError(t, errors.New("retrieving credentials timeout"), err.Error())
```

Comment:
```suggestion
		require.Error(t, err)
		require.EqualError(t, err, "retrieving credentials timeout")
```
nit. could probably just get rid of `require.Error` entirely